src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1585,7 +1597,134 @@ else if (ticket == TicketType.POST_TELEPORT) // post teleport
+            BlockPos size = endBlockPos.subtract(startBlockPos).offset(1, 1, 1);
+            // name,start,end, ignoreEntities, ignoredBlock,~~author~~,disk
+            theStructure.fillFromWorld(world,
+                    startBlockPos,
+                    size,
+                    !lv.get(end.offset).getBoolean(),
+                    ignoredblock.block == null ? null : ignoredblock.block.getBlockState().getBlock());
+            // lv6.setAuthor(lv.get(9).getString());//MC-140821
+            if (!returnnbt) {
+                if (ignoredblock.offset<lv.size()?lv.get(ignoredblock.offset).getBoolean():false) {
+                    return BooleanValue.of(structureManager.save(struident));
+                }
+            } else {
+                return new NBTSerializableValue(theStructure.save(new CompoundTag()));
+            }
+
+            return Value.TRUE;
+        });
+
+        // =====================================
+        // =======================================
+        expression.addContextFunction("load_structure_template", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerLevel world = cc.s.getLevel();
+            boolean recievnbt = (lv.get(0) instanceof NBTSerializableValue)
+                    && (((NBTSerializableValue) lv.get(0)).getTag() instanceof CompoundTag);
+            StructureTemplate thestr;
+            if (!recievnbt) {
+                String name = lv.get(0).getString();
+
+                StructureManager sm = world.getStructureManager();
+                Optional<StructureTemplate> optional;
+
+                ResourceLocation struident = ResourceLocation.tryParse(name);
+                if (name.isEmpty() || name.endsWith(":") || struident == null) {
+                    return Value.NULL;
+                }
+                optional = sm.get(struident);
+                if (!optional.isPresent()) {
+                    return Value.FALSE;
+                }
+                thestr = optional.get();
+            } else {
+                thestr = world.getStructureManager()
+                        .readStructure((CompoundTag) ((NBTSerializableValue) lv.get(0)).getTag());
+            }
+            BlockArgument start = BlockArgument.findIn((CarpetContext) c, lv, 1);
+            // name,pos,
+            // ignoreEntities,integrity,awake,noupdate,fluid,gravity,rotation,mirror
+            StructurePlaceSettings lv4 = new StructurePlaceSettings();
+            lv4.clearProcessors();
+
+            lv4.setIgnoreEntities(start.offset <lv.size()?lv.get(start.offset).getBoolean():false)
+                    .setFinalizeEntities(start.offset + 2<lv.size()?lv.get(start.offset + 2).getBoolean():false)
+                    .setKnownShape(start.offset + 3<lv.size()?lv.get(start.offset + 3).getBoolean():true)
+                    .setKeepLiquids(start.offset + 4<lv.size()?lv.get(start.offset + 4).getBoolean():false)
```

Comment:
please read and name all the options where you read arguments for clarity. 
It is very difficult to discern what is going on here - and the args don't match those available in the docs

Reply:
i add some comments for the args .